### DISCUSSION TITLE
Converting code from System.Drawing to Imagesharp

### DISCUSSION THREAD

**Discussion by Symbai** [SixLabors/ImageSharp#1767] [Q&A]
I'm using the following [constructor](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.bitmap.-ctor?view=net-5.0#System_Drawing_Bitmap__ctor_System_Int32_System_Int32_System_Int32_System_Drawing_Imaging_PixelFormat_System_IntPtr_) in System.Drawing to load an image from IntPtr (Scan0). How can I achieve this using Imagesharp?

`Bitmap newBitmap = new Bitmap(width, height, stride, PixelFormat.Format32bppArgb, (IntPtr)scan0);`

**Comment by antonfirsov** [comment]
Image.LoadPixelData with a ReadOnlySpan of pixels (made from `scan0`):
https://github.com/SixLabors/ImageSharp/blob/2b85b553f34d6ebf8ded1b390da4e9231a5e4d28/src/ImageSharp/Image.LoadPixelData.cs#L38

~Format32bppArgb matches our Rgba32 AFAIK.~ Edit: Sorry it's gonna be [Bgra32](https://github.com/SixLabors/ImageSharp/blob/2b85b553f34d6ebf8ded1b390da4e9231a5e4d28/src/ImageSharp/PixelFormats/PixelImplementations/Bgra32.cs#L13)

**Reply by Symbai** [reply]
How do I make a readonlyspan of pixels from scan0 since its a pointer and I don't know how much I should read from it.

**Reply by antonfirsov** [reply]
You need [this](https://docs.microsoft.com/en-us/dotnet/api/system.readonlyspan-1.-ctor?view=net-5.0#System_ReadOnlySpan_1__ctor_System_Void__System_Int32_) constructor to read `width * height` of `Bgra32`-s. If you are new to `Span<T>` and `ReadOnlySpan<T>`, I recommend to have a quick look at the following intros:
https://www.codemag.com/Article/1807051/Introducing-.NET-Core-2.1-Flagship-Types-Span-T-and-Memory-T
https://docs.microsoft.com/en-us/archive/msdn-magazine/2018/january/csharp-all-about-span-exploring-a-new-net-mainstay

Your code should be something like this, assuming `scan0` is a pointer:

```C#
var span = new ReadOnlySpan<Bgra32>(scan0, width * height);
using var image = Image.LoadPixelData<Bgra32>(span, width, height);
```

Note that `LoadPixelData` does not support stride with extra padding, all pixel rows in the memory area pointed by `scan0` should be exactly of length `width * height * sizeof(Bgra32)` for the above code to work.
